openstack/networking/v2/subnets/requests.go

```diff
@@ -79,7 +79,7 @@ type CreateOpts struct {
 	Name            string                `json:"name,omitempty"`
 	TenantID        string                `json:"tenant_id,omitempty"`
 	AllocationPools []AllocationPool      `json:"allocation_pools,omitempty"`
-	GatewayIP       *string               `json:"gateway_ip"`
+	GatewayIP       *string               `json:"-"`
```

Comment:
The map should have three different forms:

1. User explicitly sets a gateway: 

`map[network_id:12345 cidr:1.2.3.0/24 gateway_ip:1.2.3.1]`

2. User does not want a gateway to exist: 

`map[network_id:12345 cidr:1.2.3.0/24 gateway_ip:]`

3. User wants Neutron to create a gateway, which will end up being `1.2.3.1`: 

`map[network_id:12345 cidr:1.2.3.0/24]`